### PR TITLE
Refactor to use proxy. change api significantly.

### PR DESCRIPTION
**Pull request by alansouzati**

### CODE REVIEW

File: src/descToJSON.js [hunk 1]
```diff
@@ -68,59 +68,42 @@ const propTypeFormat = (propType, prefix = '') => {
...
     throw new Error('react-desc: component is required');
   }
 
   const documentation = {
     name: component.displayName || component.name,
   };
-  if (component.$$reactDesc) {
-    Object.assign(documentation, component.$$reactDesc);
+  if (reactDesc) {
+    Object.assign(documentation, reactDesc);
```
Comment (ericsoderberghp) [inline src/descToJSON.js]: I think Object.assign doesn't work for IE.

File: src/describe.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import PropTypes from 'prop-types';
+import descToJSON from './descToJSON';
+import descToMarkdown from './descToMarkdown';
+
+const convertArray = array =>
```
Comment (RyanCCollins) [inline src/describe.js]: Remember last week when we talked about FP, I mentioned about data last? Not that it's important to make this change, but I wanted to show you how that'd work with lodash.

```
import map from 'lodash/fp/map';

const convertArray = map(type => convertPropType(type)); // returns a function that takes the array.
```

The main reason to do things this way is to enable data transformations by composition.  Say you needed to combine like five small functions together.  With each of them curried, taking data last, you could compose them like such:

```
const doFooToBar = ...
const doBarToFoo = ...

const megaFunction = compose(convertArray, doFooToBar, doBarToFoo);

megaFunction(myData);
```

File: src/descToJSON.js (hunk 1 above)
Comment (RyanCCollins) [reply src/descToJSON.js]: This is true.  I like using spread operator for cloning.

File: __tests__/__snapshots__/describe-test.js.snap [hunk 3]
```diff
@@ -0,0 +1,345 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (RyanCCollins) [inline __tests__/__snapshots__/describe-test.js.snap]: Man you are the snapshot master :)

File: src/descToMarkdown.js [hunk 4]
```diff
@@ -1,11 +1,11 @@
 import descToJSON from './descToJSON';
 
-export default function descToMarkdown(component) {
+export default function descToMarkdown(component, reactDesc) {
   if (!component) {
```
Comment (RyanCCollins) [inline src/descToMarkdown.js]: I like the sanity checking!  I've always wondering about using facebook's invariant library for stuff like this.

**Comment by RyanCCollins** [comment]
@alansouzati Looks great man! I left a few comments.  Let me know if you want me to test anything specifically.  Otherwise, I'm confident in what I see :)

File: src/describe.js (hunk 2 above)
Comment (alansouzati) [reply src/describe.js]: thanks for spotting that. but this is not need at all anymore. PropTypes now is not an array.

Comment (alansouzati) [reply src/describe.js]: yeah. i like this idea. im getting more and more confident with FP :)

**Pull request merged**